pkg/processor/batchprocessor.go

```diff
@@ -90,22 +91,30 @@ func (s *segmentsBatch) poll() {
 			}
 			elapsed := time.Since(start)
 
+			batchesMap := make(map[string]string)
+			idRegexp := regexp.MustCompile(`\"id\":\"(.*?)\"`)
+			for i := 0; i < len(batch); i++ {
```

Comment:
Move the mapping logic into the `len() != 0` block

Reply:
fix in next version